### ISSUE TITLE
Error with fixed effects model 

### ISSUE THREAD

**Issue by kelshmo** [GabrielHoffman/variancePartition#16]
I'm running `varianceParition` 1.19.6. To repro the error from `voomWithDreamWeights`:

```
data(varPartDEdata)
countMatrix <- countMatrix[1:1000,]
gene_expression <- edgeR::DGEList(countMatrix)
gene_expression <- edgeR::calcNormFactors(gene_expression)
voom_gene_expression <- variancePartition::voomWithDreamWeights(counts = gene_expression$counts,
                                                                formula = "~ Disease + DiseaseSubtype",
                                                                data = metadata,
                                                                quiet = TRUE)
voom_gene_expression$design
```

Error:
```
Fixed effect model, using limma directly...
Error: $ operator is invalid for atomic vectors
```

**Comment by GabrielHoffman** [comment]
Your formula is really a string here.  If you remove the quotes it should work.  I will fix this internally too.

However, you will get another error since including Disease and DiseaseSubtype gives a design matrix that is not full rank.  Lets discuss tomorrow

**Comment by GabrielHoffman** [comment]
Fixed in v1.19.7.  Confirm?

**Comment by kelshmo** [comment]
Interestingly, `voom_gene_expression$design` looks good but running `voomWithDreamWeights()` returns the error message `Error: $ operator is invalid for atomic vectors`

```
> data(varPartDEdata)
> countMatrix <- countMatrix[1:1000,]
> gene_expression <- edgeR::DGEList(countMatrix)
> gene_expression <- edgeR::calcNormFactors(gene_expression)
> voom_gene_expression <- variancePartition::voomWithDreamWeights(counts = gene_expression$counts,
+                                                                 formula = "~ Disease + DiseaseSubtype",
+                                                                 data = metadata,
+                                                                 quiet = TRUE)
Error: $ operator is invalid for atomic vectors
> voom_gene_expression$design
          (Intercept) Disease1 DiseaseSubtype1 DiseaseSubtype2
sample_01           1        0               0               0
sample_02           1        0               0               0
sample_03           1        0               0               0
sample_04           1        0               0               0
sample_05           1        0               0               0
sample_06           1        0               0               0
sample_07           1        0               0               0
sample_08           1        0               0               0
sample_09           1        0               0               0
sample_10           1        0               0               0
sample_11           1        0               0               0
sample_12           1        0               0               0
sample_13           1        1               1               0
sample_14           1        1               1               0
sample_15           1        1               1               0
sample_16           1        1               1               0
sample_17           1        1               1               0
sample_18           1        1               1               0
sample_19           1        1               0               1
sample_20           1        1               0               1
sample_21           1        1               0               1
sample_22           1        1               0               1
sample_23           1        1               0               1
sample_24           1        1               0               1
attr(,"assign")
[1] 0 1 2 2
attr(,"contrasts")
attr(,"contrasts")$Disease
[1] "contr.treatment"

attr(,"contrasts")$DiseaseSubtype
[1] "contr.treatment"
```

**Comment by GabrielHoffman** [comment]
When you get the error, can you run `traceback()` and `sessionInfo()`

**Comment by kelshmo** [comment]
I revised my previous comments - should a quoted formula work in `varianceParition 1.19.7`?

**Comment by GabrielHoffman** [comment]
Your code at the beginning of this post works as is with v 1.19.7 on my machine

**Comment by kelshmo** [comment]
Ok - here is some more info to help debug:

```
> traceback()
5: terms.default(object, data = data)
4: terms(object, data = data)
3: model.matrix.default(formula, data)
2: model.matrix(formula, data)
1: variancePartition::voomWithDreamWeights(counts = gene_expression$counts, 
       formula = "~ Disease + DiseaseSubtype", data = metadata, 
       quiet = TRUE)
> sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.6

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gtools_3.8.2             sageseqr_0.0.0.9000      testthat_2.3.2          
 [4] variancePartition_1.19.7 Biobase_2.48.0           BiocGenerics_0.34.0     
 [7] scales_1.1.1             foreach_1.5.0            limma_3.44.3            
[10] ggplot2_3.3.2           

loaded via a namespace (and not attached):
  [1] backports_1.1.8         BiocFileCache_1.12.0    plyr_1.8.6             
  [4] igraph_1.2.5            splines_4.0.2           storr_1.2.1            
  [7] BiocParallel_1.22.0     usethis_1.6.1           CovariateAnalysis_0.1.0
 [10] GenomeInfoDb_1.24.2     digest_0.6.25           gdata_2.18.0           
 [13] fansi_0.4.1             magrittr_1.5            memoise_1.1.0          
 [16] base64url_1.4           cqn_1.34.0              config_0.3             
 [19] doParallel_1.0.15       remotes_2.1.1           matrixStats_0.56.0     
 [22] pack_0.1-1              askpass_1.1             drake_7.12.4           
 [25] prettyunits_1.1.1       colorspace_1.4-1        blob_1.2.1             
 [28] rappdirs_0.3.1          xfun_0.15               dplyr_1.0.0            
 [31] callr_3.4.3             crayon_1.3.4            RCurl_1.98-1.2         
 [34] roxygen2_7.1.0          lme4_1.1-23             iterators_1.0.12       
 [37] glue_1.4.1              gtable_0.3.0            zlibbioc_1.34.0        
 [40] XVector_0.28.0          MatrixModels_0.4-1      pkgbuild_1.1.0         
 [43] SparseM_1.78            DBI_1.1.0               edgeR_3.30.3           
 [46] Rcpp_1.0.5              progress_1.2.2          tmvnsim_1.0-2          
 [49] bit_1.1-15.2            mclust_5.4.6            txtq_0.2.3             
 [52] stats4_4.0.2            httr_1.4.1              gplots_3.0.4           
 [55] RColorBrewer_1.1-2      ellipsis_0.3.1          pkgconfig_2.0.3        
 [58] XML_3.99-0.4            PythonEmbedInR_0.4.65   dbplyr_1.4.4           
 [61] locfit_1.5-9.4          utf8_1.1.4              tidyselect_1.1.0       
 [64] rlang_0.4.7             reshape2_1.4.4          AnnotationDbi_1.50.1   
 [67] munsell_0.5.0           tools_4.0.2             cli_2.0.2              
 [70] generics_0.0.2          RSQLite_2.2.0           devtools_2.3.0         
 [73] stringr_1.4.0           yaml_2.2.1              processx_3.4.3         
 [76] knitr_1.29              bit64_0.9-7.1           fs_1.4.1               
 [79] synapser_0.7.64         caTools_1.18.0          purrr_0.3.4            
 [82] packrat_0.5.0           nlme_3.1-148            quantreg_5.61          
 [85] nor1mix_1.3-0           xml2_1.3.2              biomaRt_2.44.1         
 [88] compiler_4.0.2          pbkrtest_0.4-8.6        rstudioapi_0.11        
 [91] filelock_1.0.2          curl_4.3                tibble_3.0.3           
 [94] statmod_1.4.34          stringi_1.4.6           ps_1.3.3               
 [97] desc_1.2.0              lattice_0.20-41         Matrix_1.2-18          
[100] psych_1.9.12.31         nloptr_1.2.2.2          vctrs_0.3.2            
[103] pillar_1.4.6            lifecycle_0.2.0         data.table_1.12.8      
[106] bitops_1.0-6            conquer_1.0.1           GenomicRanges_1.40.0   
[109] colorRamps_2.3          R6_2.4.1                KernSmooth_2.23-17     
[112] IRanges_2.22.2          sessioninfo_1.1.1       codetools_0.2-16       
[115] boot_1.3-25             MASS_7.3-51.6           assertthat_0.2.1       
[118] pkgload_1.1.0           openssl_1.4.2           rprojroot_1.3-2        
[121] withr_2.2.0             mnormt_2.0.1            S4Vectors_0.26.1       
[124] GenomeInfoDbData_1.2.3  hms_0.5.3               grid_4.0.2             
[127] tidyr_1.1.0             minqa_1.2.4  
```

**Comment by GabrielHoffman** [comment]
I made a very small update in v1.19.8.  Give that at try.  It works for me on R 4.02 on OS X and R 3.6.0 on Linux

Can you reproduce the error on multiple machines?

**Comment by kelshmo** [comment]
This error is resolved in `1.19.8`.
